### PR TITLE
Internal: Update onboarding to previous flow [ED-25261]

### PR DESCRIPTION
**Pull request by max-zu**
## PR Checklist

- [ ] The commit message follows our guidelines:  https://github.com/elementor/elementor/blob/master/.github/CONTRIBUTING.md

## PR Type
What kind of change does this PR introduce?

- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Documentation content changes
- [ ] Other... Please describe:

## Summary

This PR can be summarized in the following changelog entry:

*

## Description
An explanation of what is done in this PR

*

## Test instructions
This PR can be tested by following these steps:

*

## Quality assurance

- [ ] I have tested this code to the best of my abilities
- [ ] I have added unittests to verify the code works as intended
- [ ] Docs have been added / updated (for bug fixes / features)

Fixes #

## 1. Problem & Context

Rolling back onboarding to the legacy (non-New) flow. The new onboarding module (`OnboardingNew`) is feature-gated behind `e_onboarding` experiment, with the old flow as default. Intent: keep both implementations parallel, gradually migrate when experiment validates.

## 2. What Changed (Where)

| Component | Change |
|-----------|--------|
| **app/app.php** | Added `OnboardingNewModule` import; conditional registration based on `e_onboarding` experiment |
| **app/modules/onboarding/module.php** | Reverted to legacy flow—280+ lines of feature-detection, AB testing, site setup logic (logo/name/theme) |
| **app/modules/onboarding/assets/js/** | Full legacy JS app: pages (account, hello-theme, site-name, site-logo, good-to-go, choose-features), utilities (tracking, storage), components (layout, buttons, forms) |
| **app/modules/onboarding-new/*** | Preserved new implementation untouched (~312 lines module.php + endpoints/storage/validation) |
| **app/assets/styles/** | Added SCSS imports for legacy onboarding UI (98 new sheets) |
| **tests/playwright/** | Force `e_onboarding` experiment active to test new flow; reset in cleanup |

## 3. How It Works

**Routing**: `app/app.php` checks `is_e_onboarding_active()` → loads `OnboardingNewModule` (new flow) or `OnboardingModule` (legacy). Both register identical component name ('onboarding'), so frontend routes `/onboarding/*` to whichever is active.

**Legacy flow**: Site setup wizard (connect account → theme selection → features/site-name/logo → "good to go"). Experiment-driven variants (101, 201, 202, 401, 402) apply A/B copy/UI changes. Event tracking defers if user unconnected; flushes on connect.

**New flow**: Minimal onboarding (theme selection + features, no site name/logo). Simplified data model, fewer endpoints, no legacy AB testing infrastructure.

Experiment flag stored in browser localStorage per variant; controls button text, step visibility, tracking behavior.

## 4. Risks

- **Experiment state mismatch**: If `e_onboarding` flag set but `OnboardingNewModule` not in classpath, silently falls back to legacy. No validation that module exists before instantiation.
- **Data migration**: No path to convert legacy onboarding progress (site name, logo) to new model if user switches experiments mid-flow.
- **Event tracking collision**: Both modules fire tracking events to same storage keys; if accidentally both load, stored events corrupt or duplicate.

**Mitigations**: Experiment flag defaults `INACTIVE` (safe default). localStorage is user-isolated (no cross-session bleed). Tests force experiment on/off explicitly.

_Generated by LinearB AI and added by gitStream._
<sub>AI-generated content may contain inaccuracies. Please verify before using.
💡 **Tip:** You can customize your AI Description using **Guidelines** [Learn how](https://docs.gitstream.cm/automation-actions/#describe-changes)</sub>

### CODE REVIEW

File: tests/playwright/sanity/core/app/modules/onboarding/onboarding.test.ts [hunk 1]
```diff
@@ -18,6 +18,9 @@ test.describe( 'Onboarding @onboarding', () => {
 		const wpAdmin = new WpAdminPage( page, testInfo, apiRequests );
 		originalActiveTheme = await wpAdmin.getActiveTheme();
 		await wpAdmin.activateTheme( 'twentytwentyfive' );
+		// This suite exercises the new onboarding flow (`e_onboarding`), which is
+		// no longer the default. Force it active so these specs keep testing it.
+		await wpAdmin.setExperiments( { e_onboarding: true } );
```
Comment (Ntnelbaba) [inline tests/playwright/sanity/core/app/modules/onboarding/onboarding.test.ts]: **Medium — default path untested in E2E**

This suite correctly forces `e_onboarding: true` so the *new* flow keeps coverage, but production default after this PR is **legacy**. CI can stay green while the shipped Connect → Hello → Features → Good-to-go path regresses.

Please add a sibling `test.describe` (or suite) with `e_onboarding: false` that walks the legacy UI and mocks `admin-ajax.php` actions instead of `/wp-json/elementor/v1/onboarding/*`.

File: app/modules/onboarding/features-usage.php [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+	 *
+	 * Expects an associative array of known tiers, each holding a list of
+	 * free-text feature labels (see assets/js/utils/utils.js), e.g.:
+	 * [ 'essential' => [ 'Templates & Theme Builder' ], 'advanced' => [], 'one' => [] ].
+	 *
+	 * @param mixed $features
+	 *
+	 * @return array|null Sanitized array, or null if the payload shape is invalid.
+	 */
+	private function sanitize_features( $features ) {
```
Comment (Ntnelbaba) [inline app/modules/onboarding/features-usage.php]: **Medium — no PHPUnit for restored features save**

`sanitize_features()` / `save_onboarding_features()` are real production code on the default path, but this PR only keeps/moves `onboarding-new` PHPUnit. Worth a small unit suite: invalid shape → error, unknown tier → reject, non-string labels → sanitize/reject, happy path → option written.

File: app/assets/js/index.js [hunk 3]
```diff
@@ -4,6 +4,7 @@ import ImportExport from '../../modules/import-export/assets/js/module';
 import ImportExportCustomization from '../../modules/import-export-customization/assets/js/module';
 import KitLibrary from '../../modules/kit-library/assets/js/module';
 import Onboarding from '../../modules/onboarding/assets/js/module';
+import OnboardingNew from '../../modules/onboarding-new/assets/js/module';
```
Comment (Ntnelbaba) [inline app/assets/js/index.js]: **Low — both onboarding modules statically imported**

Only one is constructed, but both entry graphs ship in the app bundle when `e_onboarding` is off. Optional follow-up: `import()` behind the same flag check so the unused module stays out of the default bundle.

File: app/modules/onboarding/assets/js/components/go-pro-popover.js [hunk 4]
```diff
@@ -0,0 +1,151 @@
...
+			window.open(
+				alreadyHaveProButton.href + '&mode=popup',
+				'elementorUploadPro',
+				`toolbar=no, menubar=no, width=728, height=531, top=100, left=100`,
+			);
+
+			// Run the callback for when the upload succeeds. Remove any previously
+			// registered handler first, so repeated clicks don't stack listeners.
+			elementorCommon.elements.$body
+				.off( 'elementor/upload-and-install-pro/success' )
```
Comment (Ntnelbaba) [inline app/modules/onboarding/assets/js/components/go-pro-popover.js]: **Low — listener lifecycle**

`.off().on(...)` fixes stacking on repeat clicks (thanks). The handler still lives on `elementorCommon.elements.$body` with no unmount cleanup when leaving onboarding. Prefer registering in a `useEffect` with `.off(...)` in the cleanup (or once at module scope).

File: app/modules/onboarding/module.php [hunk 5]
```diff
@@ -1,312 +1,563 @@
...
+				'url' => $custom_logo_src ? $custom_logo_src[0] : '',
 			],
+			'utms' => [
+				'connectTopBar' => '&utm_content=top-bar',
+				'connectCta' => '&utm_content=cta-button',
+				'connectCtaLink' => '&utm_content=cta-link',
+				'downloadPro' => '?utm_source=onboarding-wizard&utm_campaign=my-account-subscriptions&utm_medium=wp-dash&utm_content=import-pro-plugin&utm_term=' . self::VERSION,
+			],
+			'nonce' => wp_create_nonce( 'onboarding' ),
+			'experiment' => true,
```
Comment (Ntnelbaba) [inline app/modules/onboarding/module.php]: **Info — `experiment => true` vs `e_onboarding`**

This hardcoded key drives Features vs Site Name/Logo in JS and matches the AC features-screen default, but it is easy to confuse with the `e_onboarding` experiment. Follow-up: rename (e.g. `useFeaturesStep`) or document in module settings.

**Review by Ntnelbaba — COMMENTED** [review]
**Approve with nits** on `1628528` (ED-25261).

Restores legacy onboarding as the default and gates the March-2026 flow behind hidden experiment `e_onboarding` (default inactive). PHP (`app/app.php`) and JS (`app/assets/js/index.js`) agree. Default path matches AC: Connect → Hello-only theme (skip OK) → Features → Good-to-go. Prior sanitization / Playwright-flag concerns look addressed on tip; Pro upload listener no longer stacks on repeat click.

Nits below (test coverage for the **shipped** default path is the main follow-up). Not merge-blocking on product behavior.

**CI note:** Lint JS/PHP, Jest, QUnit, Build plugin green; some PHPUnit matrix still pending on this tip.

File: app/assets/js/index.js (hunk 3 above)
Comment (max-zu) [reply app/assets/js/index.js]: Ack — optional follow-up. Leaving static imports for now; default path still constructs only one module. Happy to do dynamic `import()` in a follow-up if we want the unused graph out of the default bundle.

File: app/modules/onboarding/module.php (hunk 5 above)
Comment (max-zu) [reply app/modules/onboarding/module.php]: Ack — follow-up. Leaving the key as-is for this PR to avoid churn; will rename/document (e.g. `useFeaturesStep`) separately if we touch this area again.

Comment (max-zu) [reply app/modules/onboarding/module.php]: Ack — leaving as follow-up (rename / document). Not addressing in this PR.

File: app/modules/onboarding/assets/js/components/go-pro-popover.js (hunk 4 above)
Comment (max-zu) [reply app/modules/onboarding/assets/js/components/go-pro-popover.js]: Ack — `.off().on` already stops stacking; full useEffect unmount cleanup as follow-up. Leaving as-is for this PR.

File: app/assets/js/index.js (hunk 3 above)
Comment (max-zu) [reply app/assets/js/index.js]: Ack — optional dynamic import follow-up. Leaving static imports as-is for this PR.

File: app/modules/onboarding/features-usage.php (hunk 2 above)
Comment (max-zu) [reply app/modules/onboarding/features-usage.php]: Ack — PHPUnit for `sanitize_features` / `save_onboarding_features` as follow-up (per Approve-with-nits). Not blocking this merge.

File: tests/playwright/sanity/core/app/modules/onboarding/onboarding.test.ts (hunk 1 above)
Comment (max-zu) [reply tests/playwright/sanity/core/app/modules/onboarding/onboarding.test.ts]: Ack — legacy default-path E2E suite as follow-up (per Approve-with-nits / not merge-blocking). Leaving for a follow-up PR.

**Review by Ntnelbaba — APPROVED** [review]
**APPROVE** on `1628528` (ED-25261).

Re-checked after Maksim's ask: tip unchanged, CI green (required checks pass). Product AC still holds — legacy onboarding default, new flow behind inactive `e_onboarding`.

Earlier nits (legacy Playwright coverage, `sanitize_features` PHPUnit, bundle/listener follow-ups) remain valid but are deferred to follow-ups as agreed; none are merge-blocking.

Prior COMMENT review: https://github.com/elementor/elementor/pull/36998#pullrequestreview-5302152351
